src/platform/site-wide/user-nav/components/SearchMenu.jsx

```diff
@@ -1,65 +1,204 @@
+import { connect } from 'react-redux';
+import { toggleValues } from 'platform/site-wide/feature-toggles/selectors';
+import FEATURE_FLAG_NAMES from 'platform/utilities/feature-toggles/featureFlagNames';
 import classNames from 'classnames';
 import recordEvent from 'platform/monitoring/record-event';
+import debounce from 'platform/utilities/data/debounce';
 
 import { replaceWithStagingDomain } from '../../../utilities/environment/stagingDomains';
 import IconSearch from '@department-of-veterans-affairs/formation-react/IconSearch';
 import DropDownPanel from '@department-of-veterans-affairs/formation-react/DropDownPanel';
 
-import Typeahead, { isTypeaheadEnabled, typeaheadListId } from './Typeahead';
-
-class SearchMenu extends React.Component {
+export class SearchMenu extends React.Component {
   constructor(props) {
     super(props);
+    this.getSuggestions = debounce(
+      this.props.debounceRate,
+      this.getSuggestions,
+    );
     this.state = {
       searchAction: replaceWithStagingDomain('https://www.va.gov/search/'),
       userInput: '',
+      suggestions: [],
     };
   }
 
-  componentDidUpdate() {
-    this.refs.searchField.focus();
+  componentDidMount = () => {
+    if (this.props.searchTypeaheadEnabled) {
+      recordEvent({
+        event: 'search-type-ahead-enabled',
+      });
+      return;
+    }
+    recordEvent({
+      event: 'search-type-ahead-disabled',
+    });
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    const { userInput } = this.state;
+
+    const inputChanged = prevState.userInput !== userInput;
+    if (inputChanged) {
+      this.getSuggestions();
+    }
+  }
+
+  async getSuggestions() {
+    const { userInput } = this.state;
+
+    if (userInput?.length <= 2) {
+      if (this.state.suggestions.length > 0) {
+        this.setState({ suggestions: [] });
+      }
+
+      return;
+    }
+
+    const encodedInput = encodeURIComponent(userInput);
+    const response = await fetch(
+      `https://search.usa.gov/sayt?=&name=va&q=${encodedInput}`,
+    );
+
+    const suggestions = await response.json();
+    this.setState({ suggestions });
   }
 
   handleInputChange = e => {
     this.setState({ userInput: e.target.value });
   };
 
+  logSearchEvent = suggestion => () => {
+    if (suggestion) {
+      recordEvent({
+        event: 'view_search_results',
+        'search-page-path': document.location.pathname,
+        'search-text-input': this.state.userInput,
+        'type-ahead-option-keyword-selected': suggestion,
+        'type-ahead-option-position':
+          this.state.suggestions.indexOf(suggestion) + 1,
+        'type-ahead-options-list': this.state.suggestions,
+      });
+      return;
+    }
+    recordEvent({
+      event: 'view_search_results',
+      'search-page-path': document.location.pathname,
+      'search-text-input': this.state.userInput,
+      'type-ahead-option-keyword-selected': suggestion,
+      'type-ahead-option-position': undefined,
```

Comment:
I'm not sure if Google Analytics tracks `undefined`, we might need to put `null` here instead for it to show up in the Google Tag Manager dashboard